src/value.rs

```diff
@@ -215,6 +217,40 @@ impl Uninitialized for Value {
     }
 }
 
+impl FromGlibPtrNone<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_none(ptr: *mut gobject_ffi::GValue) -> Self {
+        assert!(!ptr.is_null());
+        Value(ptr::read(ptr))
+    }
+}
+
+impl FromGlibPtrFull<*mut gobject_ffi::GValue> for Value {
+    #[inline]
+    unsafe fn from_glib_full(ptr: *mut gobject_ffi::GValue) -> Self {
+        let value = Value(ptr::read(ptr));
+        gobject_ffi::g_value_unset(ptr);
+        value
```

Comment:
Also it has actually the same problem, just slightly different. If you didn't have the g_value_unset(), it would be fine. But because of the g_value_unset(), for things stored as a pointer you will copy over the pointer and then immediately free it, leaving you with a pointer to a now invalid memory address.